src/processor/evict_seat.rs

```diff
@@ -117,11 +117,9 @@ pub fn process_evict_seat(program_id: &Pubkey, accounts: &[AccountInfo]) -> Prog
 
         let registered_traders = market.get_registered_traders();
 
-        if registered_traders.capacity() != registered_traders.len() && !is_fully_authorized {
-            msg!("Permissionless evictions are only allowed when the market is full");
-            // Fail silently to allow atomic seat registration following eviction failure due to market not being full
-            return Ok(());
-        }
+        // When this boolean is true, it gives the signer the privilege to evict any seat with 0 locked base lots and 0 locked quote lots
+        let has_partial_eviction_privileges =
```

Comment:
why not `has_eviction_privileges`? that's how this ends up being used. then can remove the `&& !is_fully_authorized` here and `|| is_fully_authorized` later